### PR TITLE
Feature/cr 2810 gcp db auth rule

### PR DESCRIPTION
**Pull request by idan-aviv**

### CODE REVIEW

File: cloudrail/knowledge/rules/gcp/gcp_rules_metadata.yaml [hunk 1]
```diff
@@ -87,5 +87,33 @@ rules_metadata:
...
+      process, which moves the authentication boundary from the Database Engine level to the database level,
+      hence it is recommended to disable ‘contained database authentication’ flag.
+      This recommendation is applicable to SQL Server database instances. 
+    human_readable_logic: Cloudrail will review the database instance configuration within your GCP subscription and
+      Terraform plan to ensure that SQL Server database instances have ‘contained database authentication’
+      database flag set to off.
+    remediation_steps:
+      terraform: For the google_sql_database_instance resources, ensure that ‘contained database authentication’
+        database flag is set to off for SQL Server databases.
+      console: Follow the guide at https://cloud.google.com/sql/docs/sqlserver/flags  in order to ensure that
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/rules/gcp/gcp_rules_metadata.yaml:112]: Please put <> brackets surrounding the URL link

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for sql_db in env_context.sql_database_instances:
+            if sql_db.database_version and 'SQLSERVER' in sql_db.database_version.value and \
+                    sql_db.settings and sql_db.settings.database_flags and \
+                    any(db_flag.name == 'contained database authentication' and db_flag.value == 'on'
+                        for db_flag in sql_db.settings.database_flags):
+                issues.append(
+                    Issue(
+                        f"The Google Cloud database instance `{sql_db.get_friendly_name()}` "
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py:19]: We need to use: "get_type()" at the beginning of the evidence strihg

**Review by Imanuel-Mizrahi — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py (hunk 2 above)
Comment (doronafota) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py:18]: you required to check if the 'contained database authentication' is off and your function is_flag_on checks only if it is 'on'. maybe change the function to get another parameter if the flag need to be 'on' or 'off'

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                        f"has database flag ‘contained database authentication’ set to off.",
+                        sql_db,
+                        sql_db))
+        return issues
+
+    def get_id(self) -> str:
+        return 'non_car_cloud_sql_contained_database_authentication_off'
+
+    def should_run_rule(self, environment_context: GcpEnvironmentContext) -> bool:
+        return bool(environment_context.sql_database_instances)
```
Comment (doronafota) [inline cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py:28]: should_run_rule function is not the same with every flag?

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py (hunk 2 above)
Comment (idan-aviv) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py:18]: @Imanuel-Mizrahi can you show me where it should be in the sentence?

Comment (Imanuel-Mizrahi) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_authentication_disable_rule.py:17]: @idan-aviv 
It should be like this:
f"The Google Cloud {sql_db.get_type()} `{sql_db.get_friendly_name()}` " ...

**Review by doronafota — APPROVED** [review]

**Review by Imanuel-Mizrahi — APPROVED** [review]

**Pull request merged**